micro_sam/sam_annotator/image_series_annotator.py

```diff
@@ -156,11 +185,11 @@ def next_image(*args):
         if state.amg is not None:
             state.amg.clear_state()
         state.initialize_predictor(
-            image, model_type=model_type, ndim=2, save_path=image_embedding_path,
+            image, model_type=model_type, ndim=image.ndim, save_path=image_embedding_path,
             halo=halo, tile_shape=tile_shape, predictor=predictor,
             precompute_amg_state=precompute_amg_state, device=device,
         )
-        state.image_shape = image.shape[:-1] if image.ndim == 3 else image.shape
+        state.image_shape = image.shape
```

Comment:
Hehe my bad, I was missing updating the image shape. Now it works for 3d stack series as well